app/models/image.rb

```diff
@@ -11,7 +12,12 @@ class Image < ApplicationRecord
   after_initialize :set_image_data_from_content, :if => proc { hashcode.nil? && content.present? }
 
   def set_image_data_from_content
-    self.extension = Magick::Image.from_blob(decoded_image).first&.format
+    begin
+      self.extension = Magick::Image.from_blob(decoded_image).first&.format
+    rescue StandardError
+      Rails.logger.debug("Bad image data, trying again with xml header added...")
+      self.extension = Magick::Image.from_blob(XML_HEADER + decoded_image).first&.format
```

Comment:
@lindgrenj6 This doesn't seem right, its not going to catch an exception here. Can't we just add the prefix if its missing. We can search for the XML_HEADER if missing we prepend it then pass it to Magic::Image.from_blob.
Here any StandardError that is thrown we will catch and interpret it to mean that the XML_HEADER is missing and try it by prepending the XML_HEADER. Can't topology add the XML_HEADER if it's missing when they do an inventory refresh. That way the change will be in one place in the Topology. If the valid definition of an SVG content is to have the XML_HEADER it makes sense to add this in topology. Maybe we should fix it topology and create a PR on that repo.

Reply:
@mkanoor the XML_HEADER is only added so `ImageMagick` can use a string library to validate svg.  It isn't needed for a browser or any other operation. 

We don't want to prepend the xml header in all cases because if the blob is a jpg, png, then the xml_header is not needed as ImageMagick uses a binary validation library. In those cases it would return a false for a valid jpg, png.

Reply:
@lindgrenj6 @syncrou 
Thanks for the clarification 👍